File: packages/sdk-go/settler.go
package settler

import (
	"context"
	"os"
)

// Config holds configuration for the Settler client
type Config struct {
	APIKey  string
	BaseURL string
}

// Client is the main client for interacting with Settler API
type Client struct {
	config Config
}

// NewClient creates a new Settler client
func NewClient(config Config) *Client {
	if config.APIKey == "" {
		config.APIKey = os.Getenv("SETTLER_API_KEY")
	}
	if config.BaseURL == "" {
		config.BaseURL = "https://api.settler.dev"
	}
	return &Client{config: config}
}

// ReconcileRequest represents a reconciliation request
type ReconcileRequest struct {
	Source interface{} `json:"source"`
	Target interface{} `json:"target"`
	Rules  interface{} `json:"rules"`
}

// Reconcile reconciles transactions
func (c *Client) Reconcile(ctx context.Context, req ReconcileRequest) (interface{}, error) {
	// Implementation will be added from private repo
	panic("not implemented yet")
}

File: packages/sdk-ruby/lib/settler.rb
# frozen_string_literal: true

require 'faraday'
require 'json'

module Settler
  # Client for Settler API
  class Client
    def initialize(api_key:, base_url: nil)
      @api_key = api_key || ENV['SETTLER_API_KEY']
      @base_url = base_url || 'https://api.settler.dev'
      
      raise ArgumentError, 'API key is required' if @api_key.nil?
    end

    # Reconcile transactions
    def reconcile(data)
      # Implementation will be added from private repo
      raise NotImplementedError, 'Not implemented yet'
    end
  end
end
